### PR TITLE
DM-35191: Add test matrix for python versions

### PR DESCRIPTION
**Pull request by mwittgen**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -77,7 +81,7 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v3
         with:
-          python-version: "3.10"
+          python-version: "3.9"
```
Comment (timj) [inline .github/workflows/build.yaml:84]: Why is this now 3.9?

**Review by timj — APPROVED** [review]
One question.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (mwittgen) [reply .github/workflows/build.yaml:84]: 3.9 and 3.10 pass the matrix test. Would we pick the lowest python version for publishing? Or just use 3.10 and 
also declare 3.10 required?

Comment (mwittgen) [reply .github/workflows/build.yaml]: Changed it back to 3.10

**Pull request merged**